### PR TITLE
analysis (computed) column improvements

### PR DESCRIPTION
**Pull request by JorisGoosen**
fix for jasp-stats/jasp-test-release#2340

columntype for analysis created computed columns cannot be changed anymore
make columnname uneditable for analysis created columns
make sure analysisform is shown

I think Columns added from analyses (not computed, such as in Distributions) are broken though.
Or am I doing something wrong with that @Kucharssim ?

### CODE REVIEW

File: Desktop/components/JASP/Widgets/DataTableView.qml [hunk 1]
```diff
@@ -679,7 +679,10 @@ FocusScope
 								filterWindow.showEasyFilter = true
 								filterWindow.open()
 							}
-						}
+                            
+                            if(!columnModel.nameEditable && columnModel.columnName != "") // so probably (computed) column from an analysis so try to show it  
+                                computedColumnsInterface.showAnalysisFormForColumn(columnModel.columnName)
+                        }
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/DataTableView.qml:685]: Why don't you use `computedColumnType == computedColumnTypeAnalysis`

File: Desktop/components/JASP/Widgets/VariablesWindow.qml [hunk 2]
```diff
@@ -53,8 +53,18 @@ FocusScope
...
+            
+            //if(!columnModel.nameEditable && columnModel.columnName != "") // so probably (computed) column from an analysis so try to show it
+            //    computedColumnsInterface.showAnalysisFormForColumn(columnModel.columnName)
 		}
+        
+        /*function onVisibleChanged(visible)
+        {
+            if(visible && !columnModel.nameEditable && columnModel.columnName != "") // so probably (computed) column from an analysis so try to show it
+                computedColumnsInterface.showAnalysisFormForColumn(columnModel.columnName)   
+        }*/
```
Comment (boutinb) [inline Desktop/components/JASP/Widgets/VariablesWindow.qml]: Can be removed probably...

**Review by boutinb — CHANGES_REQUESTED** [review]
JASP crashes if you close the analysis with a computed column. I have added a commit for this.

**Comment by JorisGoosen** [comment]
@RensDofferhoff you have time to look at this?

File: Desktop/components/JASP/Widgets/DataTableView.qml (hunk 1 above)
Comment (JorisGoosen) [reply Desktop/components/JASP/Widgets/DataTableView.qml:685]: Yeah good point

**Comment by RensDofferhoff** [comment]
@JorisGoosen I will take a look

**Review by boutinb — APPROVED** [review]

**Review by RensDofferhoff — DISMISSED** [review]
The column data from samples generated in distributions is not shown. But thats probably an unrelated minor issue (not sure if it ever did)

**Review by RensDofferhoff — COMMENTED** [review]
one second I might still have a crash

**Comment by RensDofferhoff** [comment]
It still crashes when closing with an analysis generated column

**Comment by JorisGoosen** [comment]
> It still crashes when closing with an analysis generated column

"Closing" what?
Do you mean deleting? Cause that doesnt crash it here, neither does closing the variableswindow.

**Comment by RensDofferhoff** [comment]
> > It still crashes when closing with an analysis generated column
> 
> "Closing" what? Do you mean deleting? Cause that doesnt crash it here, neither does closing the variableswindow.

Close the dataset in the filemenu

**Comment by JorisGoosen** [comment]
Aah that is not good :p

Edit: And I can easily reproduce this...

**Review by RensDofferhoff — APPROVED** [review]

**Pull request merged**